helm-charts/basehub/values.yaml

```diff
@@ -5,6 +5,14 @@ userServiceAccount:
   enabled: true
   annotations: {}
 
+ingressBasicAuth:
+  enabled: false
+  # Primarily here for validation to 'work',
+  # as these are set in secret config otherwise. I don't like this,
+  # as we won't catch these values missing if they aren't set.
+  username: ""
+  password: ""
```

Comment:
This can be handled in the helm templates rendering conditionally on `.Values.ingressBasicAuth.enabled` by having a pipe after the reference to `.Values.ingressBasicAuth.username` like this: `| required "ingressBasicAuth.username is required"`

So for example you can have something like this in the template.

```yaml
  auth: {{ htpasswd (.Values.ingressBasicAuth.username | required "ingressBasicAuth is enabled, a username is required" ) (.Values.ingressBasicAuth.password | required "ingressBasicAuth is enabled, a username is required") | b64enc | quote}}
{{- end }}
```

Another option is to let this be validated in the jsonschema, but no matter what this will end up being caught during helm template rendering.

Reply:
An example like this can be found in z2jh here: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/034c3a4ec71710dbba2d605361187003156d6b92/jupyterhub/values.schema.yaml#L81-L173

Reply:
@consideRatio I'd like to do this in a follow-up PR, given that the ESA hub needs to be out shortly. We also already use the empty string trick elsewhere, so should probably be fixed alltogether.

Reply:
I opened https://github.com/2i2c-org/infrastructure/issues/2494